sky/shell/tracing_controller.cc

```diff
@@ -89,14 +103,33 @@ static const char* ObservatoryInvoke(const char* method,
 TracingController::TracingController()
     : picture_tracing_enabled_(false),
       terminate_loop_on_write_(false),
+      dart_initialized_(false),
+      tracing_active_(false),
       weak_factory_(this) {
   ManageObservatoryCallbacks(true);
+
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
```

Comment:
Why not just do this in the caller?  Then this file doesn't need to know about commandline, etc.

Also, I think we have a switches.cc where these switches should be defined.